Getting rid of bare excepts where possible

ayodejiige opened the pull request:
The last exception in _print_RandomDomain() should be called when the
RandomDomain object doesn't have a set property which will raise an
IndexError

<If this pull request fixes an issue please indicate which issue by typing "Fixes #NNNN" below.>
This is a first commit for #11668

ayodejiige committed:
exceptions: Changed bare exception in str.py to explicit exception

The last exception in _print_RandomDomain() should be called when the
RandomDomain object doesn't have a set property which will raise an
IndexError

ayodejiige committed:
Replaced bare exceptions in printer module with explicit ones

The _print_RandomDomain() function in the str, pretty and latex classes
had exceptions that would be raised by lack of certain properties as a
result of what RandomDomain subclass the object being printed belonged to.

asmeurer commented:
Wouldn't the second case also raise AttributeError?

ayodejiige commented:
It raises an index error because the RandomDomain class and child classes have a set property.
` @property
    def set(self):
        return self.args[1]`
The index error will be raised in cases where self.args[1] doesn't exist.

ayodejiige commented:
I did notice that I wouldn't this case though:
```python
class ConditionalDomain(RandomDomain):
    ...
    @property
    def set(self):
        raise NotImplementedError("Set of Conditional Domain not Implemented")
   ...
```
I will add the code to catch that. Also, this is my first open source project and I was wondering if issues on this projects are typically fixed with one pull request and if it's fine to continuously push to a pull request.

asmeurer commented:
Please push to the same pull request.

asmeurer commented:
It would be better to refactor the code to not use try-except logic at all. The fact that `set` raises IndexError is not at all obvious and could change if the implementation changes. It would be better to do `hasattr` checks.

ayodejiige commented:
I will make the necessary changes. Thanks!

ayodejiige committed:
Repalced exceptions in printing module and modified bare exceptions in
matrices module

Replaced exception in _print_RandomDomain() function of the printing
module with if-else logic. Also modified bare except in classof() function
of matrices module

gxyd commented:
@asmeurer can you please see if the changes are apt? Seems like he did 'try' address the requested changes.

gxyd commented:
@nilabja10201992 would you like to review this pull request?

nilabja10201992 commented:
@gxyd sure Sure I'll review them btw should I work on the fixes that you mentioned yesterday?

gxyd commented:
Yes, you can work on that as well.

nilabja10201992 commented:
Okay :)

asmeurer commented on sympy/printing/pretty/pretty.py:
```diff
@@ -1975,20 +1975,22 @@ def _print_KroneckerDelta(self, e):
+            pform = prettyForm(*pform.right(self._print(d.symbols)))
+            return pform
+        else:
+            return None
```
I believe this should be `self._print(None)` to match what was there before.

asmeurer commented on sympy/printing/str.py:
```diff
@@ -118,14 +118,15 @@ def _print_Dict(self, expr):
+        elif hasattr(d, 'symbols'):
+            return 'Domain on ' + self._print(d.symbols)
+        else:
+            return None
```
Printers should not return None. The above case should be the `else` case.

asmeurer commented on sympy/printing/latex.py:
```diff
@@ -1564,14 +1564,15 @@ def _print_ProductSet(self, p):
+        elif hasattr(d, 'symbols'):
+            return 'Domain on ' + self._print(d.symbols)
+        else:
+            return None
```
Same here.

nilabja10201992 commented:
@gxyd seems correct to me, as @asmeurer stated _else shouldn't return None_, otherwise, everything seems okay.

gxyd commented:
@ayodejiige can you please address the requested changes?

gxyd commented:
@nilabja10201992 would you like to make changes here as suggested by Aaron?
You would need to add 'remote' here. The main purpose could be figuring out how to push to some other 'remote', search for it over the web.

In any case, let me know if you need help.

nilabja10201992 commented:
@gxyd I've made changes to my local branch but cannot push to the remote branch. It says access denied.

gxyd commented:
I'll get back in the morning.

gxyd commented:
Can you paste output of `git remote -v` (running in SymPy root folder) here?

nilabja10201992 commented:
```
origin	https://github.com/nilabja10201992/sympy (fetch)
origin	https://github.com/nilabja10201992/sympy (push)
pb	https://github.com/ayodejiige/sympy (fetch)
pb	https://github.com/ayodejiige/sympy (push)
upstream	https://github.com/sympy/sympy (fetch)
upstream	https://github.com/sympy/sympy (push)
```

gxyd commented:
And first you do `git remote fetch ayodejiige`. And after that you do `git checkout -b 11668_bare_excepts ayodejiige/11668_bare_excepts` (may be you **didn't** execute this command)?

And make changes to `11668_bare_excepts` **not** to `ayodejiige/11668_bare_excepts`. And finally you do `git push ayodejiige 11668_bare_excepts` (may be you did this right).

ylemkimon commented:
@gxyd Only those with write access to main branch(`sympy/sympy`) can push commits to PR branch.

nilabja10201992 commented:
@gxyd when I'm trying to push using `git it push ayodejiige 11668_bare_excepts` I'm getting these errors
```
error: src refspec 11668_bare_excepts does not match any.
error: failed to push some refs to 'ayodejiige'

```

gxyd commented:
Okay, yes you can't do that. I think that can be done via `git cherry-pick ...` command https://git-scm.com/docs/git-cherry-pick . That would be sending in a new pull request, not pushing to this branch.

nilabja10201992 commented:
@gxyd can you please look at the changes here https://github.com/sympy/sympy/compare/master...nilabja10201992:pb?expand=1 , should I pull request?

gxyd committed:
return self._print(None) instead of just 'None'

gxyd commented:
I have fixed the requested changes. Will merge it when it passes the test.

gxyd committed:
Merge branch 'master' into 11668_bare_excepts

gxyd committed:
raise 'ImportError' in case of not installed NumPy

gxyd commented:
Tests pass. Merging.
